packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -101,6 +101,8 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
     // Save the timeout so we can cancel and reschedule it as needed
     private blobCacheTimeout: ReturnType<typeof setTimeout> | undefined;
+    // If the defer flag is set when the timeout fires, we'll reschedule rather than clear immediately
+    private deferBlobCacheClear: boolean = false;
```

Comment:
Some comment somewhere describing why this complexity is needed would be nice, as next developer may not have enough context and "simplify" it :)